### PR TITLE
Debug: limit "object as children" error to elements

### PR DESCRIPTION
**Pull request by developit**
This should fix preactjs/preact-render-to-string#254 without incurring the performance overhead of converting primitives in JSX text positions to VNodes.

If this works, we can remove [these lines](https://github.com/preactjs/preact-render-to-string/blob/fac154450e5455446c1cf844f2e7cbb9e0bd4127/src/index.js#L140-L147) from render-to-string.

### CODE REVIEW

**Review by marvinhagemeister — APPROVED** [review]
Not sure why I went the other way. Agree that this is a better approach 👍

File: debug/src/debug.js [hunk 1]
```diff
@@ -318,12 +318,9 @@ export function initDebug() {
 		// Putting this check in `options.diffed` ensures that
 		// `vnode._children` is set and that we only validate the children
 		// that were actually rendered.
-		if (vnode._children) {
+		if (typeof vnode.type === 'string' && vnode._children) {
```
Comment (marvinhagemeister) [inline debug/src/debug.js:321]: Wondering if we want to only do these checks on DOM nodes or also components.

Comment (developit) [reply debug/src/debug.js:321]: Ah good point - I was forgetting that `._children` is the materialized children, not `props.children` as passed to a component. Will remove.

**Comment by developit** [comment]
Alright, I just manually confirmed that this change addresses preactjs/preact-render-to-string#254. Will open a PR there.

**Pull request merged**